src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -108,7 +108,18 @@ SourceResultType PhysicalTableScan::GetData(ExecutionContext &context, DataChunk
 	if (g_state.in_out_final) {
 		function.in_out_function_final(context, data, chunk);
 	}
-	function.in_out_function(context, data, g_state.input_chunk, chunk);
+	switch (function.in_out_function(context, data, g_state.input_chunk, chunk)) {
+	case OperatorResultType::BLOCKED: {
+		auto guard = g_state.Lock();
+		g_state.BlockSource(guard, input.interrupt_state);
```

Comment:
This has a return value, based on whether blocking is prevented or not, that should probably be respected

Reply:
Keep in mind that with this change, returning BLOCKED requires additional work on the callers side (the in-out function implementation) to prevent a deadlock - to ensure this task gets added back to the task queue

Reply:
Thanks for the comments!

Reply:
First comment handled (I think, but might pass by and ask).
Second comment: yes, I am aware, will make it clear with a test.